nautobot_circuit_maintenance/models.py

```diff
@@ -267,6 +269,13 @@ def to_csv(self):
         """Return fields for bulk view."""
         return (self.subject, self.provider, self.sender, self.source, self.raw, self.date, self.parsed)
 
+    def clean(self):
+        """Add validation when creating a RawNotification."""
+        super().clean()
+
+        if self.date < self.stamp:
```

Comment:
Shouldn't this be something like `if self.stamp > datetime.now(timezone.utc)` instead? `self.date` can be changed to any arbitrary date.

Also I hope that Nautobot is always being run on an NTP-managed system with this change, I would hate to see it start throwing errors just because of a few seconds of clock drift resulting in it receiving perfectly valid notifications "from the future"...!

Reply:
Also it doesn't look like you have test coverage for this check. :-)